builtins/amp-ad.js

```diff
@@ -283,6 +283,10 @@ export function installAd(win) {
      * @private
      */
     noContentHandler_() {
+      if (!this.placeholder_) {
+        this.element.style.height = 0;
```

Comment:
Thinking this through again. The only way to adjust scroll position is to go back to, initially, setting height to 0. Not directly, but rather via `this.changeHeight` API. What we need to do is schedule `display:none` via `deferMutate` right after `changeHeight` has completed. We currently don't have that option - so we need to add it. E.g. we need to add an optional callback to `changeHeight` API that will be called as soon as it has completed. So, it will look like this:

```
this.changeHeight(0, () => {
  this.deferMutate(() => display = 'none');
})
```

However, one more thing, 
```
if (!this.placeholder_) {
    this.element.style.display = 'none';
} else {
    this.toggleFallback(true);
}
```
I still don't think that this is correct. Placeholder and fallback are not necessarily related. There's some confusion going on here. For instance, if there's no placeholder, but there's fallback, should we still hide the whole element?